subworkflows/local/remove_redundancy.nf

```diff
@@ -0,0 +1,98 @@
+/*
+    REMOVAL OF REDUNDANT SEQUENCES AND FAMILIES
+*/
+
+include { EXTRACT_FAMILY_REPS       } from '../../modules/local/extract_family_reps.nf'
+include { CAT_CAT                   } from '../../modules/nf-core/cat/cat'
+include { HMMER_HMMSEARCH           } from '../../modules/nf-core/hmmer/hmmsearch/main'
+include { REMOVE_REDUNDANT_FAMS     } from '../../modules/local/remove_redundant_fams.nf'
+include { FILTER_NON_REDUNDANT_HMMS } from '../../modules/local/filter_non_redundant_hmms.nf'
+include { EXECUTE_CLUSTERING        } from '../../subworkflows/local/execute_clustering'
+include { REMOVE_REDUNDANT_SEQS     } from '../../modules/local/remove_redundant_seqs.nf'
+include { ALIGN_SEQUENCES           } from '../../subworkflows/local/align_sequences'
+
+workflow REMOVE_REDUNDANCY {
+    take:
+    msa   // tuple val(meta), path(fas)
+    fasta // tuple val(meta), path(fasta)
+    hmm   // tuple val(meta), path(hmm)
+
+    main:
+    ch_versions = Channel.empty()
+
+    if (params.remove_family_redundancy) {
+        msa
+            .map { meta, aln -> [[id: meta.id], aln] }
+            .groupTuple(by: 0)
+            .set { ch_msa }
+        EXTRACT_FAMILY_REPS( ch_msa )
+        ch_versions = ch_versions.mix( EXTRACT_FAMILY_REPS.out.versions )
+
+        hmm
+            .map { meta, model -> [[id: meta.id], model] }
+            .groupTuple(by: 0)
+            .set { ch_hmm }
+        CAT_CAT( ch_hmm )
+        ch_versions = ch_versions.mix( CAT_CAT.out.versions )
+
+        CAT_CAT.out.file_out
+            .combine(EXTRACT_FAMILY_REPS.out.fasta, by: 0)
+            .map { meta, model, seqs -> [meta, model, seqs, false, false, true] }
+            .set { ch_input_for_hmmsearch }
+
+        HMMER_HMMSEARCH( ch_input_for_hmmsearch )
+        ch_versions = ch_versions.mix( HMMER_HMMSEARCH.out.versions )
+
+        fasta
+            .map { meta, fas -> [[id: meta.id], fas] }
+            .groupTuple(by: 0)
+            .set { fasta }
```

Comment:
Again, please check this everywhere -use `ch_<name>` notation instead of `.set` for consistency